api/organisations/chargebee/serializers.py

```diff
@@ -0,0 +1,32 @@
+from rest_framework.exceptions import ValidationError
+from rest_framework.serializers import CharField, Serializer
+
+
+class PaymentFailedInvoiceSerializer(Serializer):
+    dunning_status = CharField(allow_null=False)
+    subscription_id = CharField(allow_null=False)
+
+    def validate_dunning_status(self, value):
+        if value != "in_progress":
+            raise ValidationError("To be valid dunning must be in progress")
+        return value
+
+
+class PaymentSucceededInvoiceSerializer(Serializer):
+    subscription_id = CharField(allow_null=False)
```

Comment:
The code is definitely much cleaner for using these serializers, but I'm not entirely sure why we need different serializers for the different event types, couldn't we use a single serializer here?

Reply:
Given that only the payment failed webhook has the `dunning_status` field I think it's better to separate out the serializers. I generally prefer smaller serializers over ones which need to introspect to determine the event type in order to enforce different policies.

Reply:
Yep, ok fair enough.